ctapipe/ml/sklearn.py

```diff
@@ -0,0 +1,137 @@
+"""
+Component Wrappers around sklearn models
+"""
+from inspect import isabstract
+from importlib import import_module
+
+import numpy as np
+from traitlets import Dict, List, Unicode, Enum, Integer
+from sklearn.base import is_classifier, is_regressor
+import joblib
+
+from ..core import Component
+from .preprocessing import check_valid_rows, table_to_float
+
+
+sklearn_modules = {
+    name: import_module(f"sklearn.{name}")
+    for name in ["ensemble", "linear_model", "neighbors", "svm", "tree", "naive_bayes"]
+}
+
+
+def get_models(filter_func=is_classifier):
+    models = {}
+    for module in sklearn_modules.values():
+        for cls_name in dir(module):
+            cls = getattr(module, cls_name)
+            if filter_func(cls) and not isabstract(cls):
+                models[cls.__name__] = cls
+
+    return models
+
+
+SUPPORTED_CLASSIFIERS = get_models(is_classifier)
+SUPPORTED_REGRESSORS = get_models(is_regressor)
+SUPPORTED_MODELS = {**SUPPORTED_CLASSIFIERS, **SUPPORTED_REGRESSORS}
+
+
+class Model(Component):
+    features = List(Unicode(), help="Features to use for this model").tag(config=True)
+    model_config = Dict({}, help="kwargs for the sklearn model").tag(config=True)
+    model_cls = Enum(SUPPORTED_MODELS.keys(), default_value=None, allow_none=False).tag(
+        config=True
+    )
+    target = Unicode(None, allow_none=False, help="Name of target column").tag(
+        config=True
+    )
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.model = SUPPORTED_MODELS[self.model_cls](**self.model_config)
+
+    def table_to_X(self, table):
+        feature_table = table[self.features]
+        valid = check_valid_rows(feature_table, log=self.log)
+        X = table_to_float(feature_table[valid])
+        return X, valid
+
+    def table_to_y(self, table):
+        return np.array(table[self.target])
```

Comment:
My gut tells me there should be a dtype here as is done in `table_to_float`. But thinking a little about it that needs to default to float32 for regressors and int for classifiers? Or should the user make sure the correct dtype is supplied? Or is sklearn smart enough that the dtype needs only to be set for `X`? Also, while typing this, I wonder about categorical variables.. but this is maybe not needed for ctapipe? Sorry for this unstructured braindump

Reply:
It needs to be int for a classifier and float something for a regressor. The dtypes are actually different from model to model depending on the underlying implementation.

Reply:
So we cannot "guess" here and the user has to make sure it works for their usecase, correct?

Reply:
I think it is safe for now to assume that it is a suitable dtype in the table, yes